src/hello-visualization/line-chart/app_1.js

```diff
@@ -0,0 +1,216 @@
+/* eslint-env browser */
+
+import Halyard from 'halyard.js';
+import angular from 'angular';
+import enigma from 'enigma.js';
+import enigmaMixin from 'halyard.js/dist/halyard-enigma-mixin';
+import qixSchema from 'enigma.js/schemas/3.2.json';
+import template from './app_1.html';
+import Linechart from './linechart';
+
+const halyard = new Halyard();
+
+angular.module('app', []).component('app', {
+  bindings: {},
+  controller: ['$scope', '$q', '$http', function Controller($scope, $q, $http) {
+    $scope.dataSelected = false;
+    $scope.showFooter = false;
+
+    $scope.toggleFooter = () => {
+      $scope.showFooter = !$scope.showFooter;
+      if (!$scope.showFooter && $scope.dataSelected) {
+        this.clearAllSelections();
+      }
+    };
+
+    $scope.openGithub = () => {
+      window.open('https://github.com/qlik-oss/core-get-started');
+    };
+
+    this.connected = false;
+    this.painted = false;
+    this.connecting = true;
+
+    let object = null;
+    let app = null;
+
+    const select = (value) => {
+      app.getField('Movie').then((field) => {
+        field.select(value).then(() => {
+          $scope.dataSelected = true;
+          this.getMovieInfo().then(() => {
+            $scope.showFooter = true;
+          });
+        });
+      });
+    };
+
+    const linechart = new Linechart();
+
+    const paintChart = (layout) => {
+        linechart.paintLinechart(document.getElementById('chart-container'), layout, {
+        select,
+        clear: () => this.clearAllSelections(),
+        hasSelected: $scope.dataSelected,
+      });
+      this.painted = true;
+    };
+
+    this.generateGUID = () => 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+      // eslint-disable-next-line no-bitwise
+      const r = Math.random() * 16 | 0;
+      // eslint-disable-next-line no-bitwise
+      const v = c === 'x' ? r : ((r & 0x3) | 0x8);
+      return v.toString(16);
+    });
+
+    this.$onInit = () => {
+      const config = {
+        Promise: $q,
+        schema: qixSchema,
+        mixins: enigmaMixin,
+        url: `ws://${window.location.hostname}:19076/app/${this.generateGUID()}`,
+      };
+
+      // Add local data
+      const filePathMovie = '/data/movies.csv';
+      const tableMovie = new Halyard.Table(filePathMovie, {
+        name: 'Movies',
+        fields: [{ src: 'Movie', name: 'Movie' }, { src: 'Year', name: 'Year' },
+          { src: 'Adjusted Costs', name: 'Adjusted Costs' }, { src: 'Description', name: 'Description' }, { src: 'Image', name: 'Image' }],
+        delimiter: ',',
+      });
+      halyard.addTable(tableMovie);
+
+      // Add web data
+      $http.get('https://gist.githubusercontent.com/carlioth/b86ede12e75b5756c9f34c0d65a22bb3/raw/e733b74c7c1c5494669b36893a31de5427b7b4fc/MovieInfo.csv')
```

Comment:
could we use async/await?

Reply:
We could, but I think that should be fixed in a separate PR. It's some work to do in both the line chart and the scatter plot examples. And some transpiling stuff.. :)

Reply:
I tried once to switch to async/await but failed. There were problems with Angular and the Promise lib used. Can't remember the details. It was not trivial,  but probably worth a new try in a separate PR.